### ISSUE TITLE
Nexus5 sim recognition and cellular data carrier selection error

### ISSUE THREAD

**Issue by necoli1822** [ubports/ubuntu-touch#405]
### Description of the feature

On Nexus5 (build 15.04 stable, rc and devel, too) cellular network carrier selection takes too long, but also doesn't work properly. I`m working on Nexus5 with sim card provided by KT olleh from Korea, Republic of.
### Illustrations

- Device: hammerhead
- Channel: stable
- Build: r3

### Steps to reproduce

On Settings > Cellular Network > Carrier and APN > Carrier > though it showes carriers just same as when I run the script in /usr/share/ofono/scripts/get-operators and scan-for-operators. 

### Expected behavoir

I want my Nexus5 work with selecting operators to make calls and smses.

### Actual behavior

Does not work both selecting operators and also sim network, too.

### Logfiles and additional information

by using Logviewer, I have got those messages below.
--START_OF_LOG--
Loading module:
'libubuntu_application_api_touch_mirclient.so.3.0.0'
qml: Page_QMLTYPE_45(0x1ea6040, "systemSettingsPage")"시스템 설정": In Ubuntu.COmponenets
1.3, the use of Page.title, Page.flickable and Page.head is deprecated. Use Page.header and the PageHeader component instead.
BrightnessItem::BrightnessItem(const QVariantMap& ,QObject*) ubuntu.widi.supported: 1
QProcess::start: Process is already running library "/vendor/lib/egl/libGLESv2S3D_adreno.so" not found
qml: PageComponent_QMLTYPE_98(0x230f1f0, "cellularPage")"셀룰러 네트워크": In Ubuntu.Components 1.3, the use of Page.title, Page.flickable and Page.head is deprecated. Use Page.header and the PageHeader component instead.
file://usr/share/ubuntu/settings/system/qml-plugins/cellular/PageComponent.qml:30:1: QML PageComponent:
Binding loop detected for property "flickable"
[PERFORMANCE]: Last frame took 46 ms to render.
file:///usr/share/ubuntu/settings/system/qml-plugins/cellular/Components/NoSim.qml:26: TypeError: Cannot read property of null
file:///usr/share/ubuntu/settings/system/qml-plugins/cellular/Components/NoSim.qml:27: TypeError: Cannot read property of null 
file:///usr/share/ubuntu/settings/system/qml-plugins/cellular/Components/NoSim.qml:28: TypeError: Cannot read property of null
qml: PageCarrierAndApn_QMLTYPE_121(0x21dfc40,"carrierApnPage")"통신사 & APN": In Ubuntu.Components 1.3, the use of Page.title, Page.flickable and Page.head is deprecated. Use Page.header and the PageHeader component instead.
qml: PageChooseCarrier_QMLTYPE_122(0x2415e20, "chooseCarrierPage")"통신사": In Ubuntu.Components 1.3, the use of Page.title, Page.flickable and Page.head is deprecated. Use Page.header and the PageHeader component instead.
QObject::startTimer: Timers cannot be started from another thread
QObject::killTimer: Timers cannot be stopped from another thread
QObject::startTimer: Timers cannot be started from anothe thread
file:///usr/lib/arm-linux-nueabihf/qt5/qml/Ubuntu/Components/1.3/OptionSelectorDelegate.qml:324:23:
Unable to assign QOfonoNetworkOperator to QString
file:///usr/lib/arm-linux-nueabihf/qt5/qml/Ubuntu/Components/1.3/OptionSelectorDelegate.qml:324:23:
Unable to assign QOfonoNetworkOperator to QStringfile:///usr/lib/arm-linux-nueabihf/qt5/qml/Ubuntu/Components/1.3/OptionSelectorDelegate.qml:324:23:
Unable to assign QOfonoNetworkOperator to QString
file:///usr/lib/arm-linux-nueabihf/qt5/qml/Ubuntu/Components/1.3/OptionSelectorDelegate.qml:324:23:
Unable to assign QOfonoNetworkOperator to QString
[PERFORMANCE]: Last frame took 41 ms to render.
QObject::killTimer: Timers cannot be stopped from another thread
QObject::startTimer: Timers cannot be stopped from another thread
QObject::killTimer: Timers cannot be stopped from another thread
QObject::startTimer: Timers cannot be stopped from another thread
QOfonoNetworkOperator::registerOperator() failed: QDBusError("org.ofono.Error.Failed", "Operation failed")
qml: Register complete: org.ofono.Error.Failed Operation failed
qml: Falling back to default operator.
QObject::killTimer:Timers cannot be stopped from another thread
QObject::startTimer:Timers cannot be stopped from another thread
qml: PageCarrierAndApn_QMLTYPE_121(0x2248728, "carrierApnPage")"통신사 & APN": In
Ubuntu.Component 1.3, the use of Page.title, Page.flickable and Page.head is deprecaterd. Use Page.header and the PageHeader component instead.
QObject::killTimer: Timers cannot be stopped from another thread
QObject::startTimer: Timers cannot be stopped from another thread
--END_OF_LOG--

**Comment by necoli1822** [comment]
Also, I found a similar error report on the launchpad (https://bugs.launchpad.net/ubuntu/+source/ubuntu-system-settings/+bug/1608387); saying that it happened on OTA12, but gone by updating to OTA13.

**Comment by LightofAeons** [comment]
Can also add I am on 15.04 (r3) with my Nexus 5 and the "carrier" option is always set to none, even though I had selected my SIM card's carrier (I'm using NTT DOCOMO on 4G) in Japan and entered the correct APN settings. I am getting the same result as Necoli1822.

**Comment by LightofAeons** [comment]
@necoli1822 I have a feeling these two issues might be linked:

https://github.com/ubports/ubuntu-touch/issues/377

I will try switching to the Dev channel and back to Stable when I get home and report if that fixes the issue.

**Comment by LightofAeons** [comment]
Switched from 15.04 Stable (r3) to 15.04 Devel (285) --> Mobile Carrier failed to be recognised.
Switched back to 15.04 Stable (r3) from 15.04 Devel (285) --> Mobile Carrier failed to be recognised.

Switched 15.04 rc --> Mobile Carrier failed to be recognised.
Switched back to 15.04 Devel (285) from 15.04 rc --> Mobile Carrier failed to be recognised.

In addition, when looking under "Network" Settings in the Drop-down menu, my Nexus 5 is always coming up with "Searching" for a signal and then after 5 minutes says "Denied". My APN settings are correct and I have checked them multiple times. The only other APN setting that is required by my provider (YNMobile, which piggyback on one of the main providers in Japan, NTT DOCOMO) is selecting the PAP or CHAP option. However, selecting PAP or CHAP isn't currently available for the Nexus 5. The only options that I have available are: Name, APN, Username, Password.

I have also tried this fix provided by AlanGriffiths:

Close "System Settings" app
Install & open "UT Tweak Tool"
Go to "Apps Scopes" -> "System Settings"
Select "Clear..." and "Cache", "Application data" and "Config" (may not be all necessary, but this worked)
Select "OK"
Open "System Settings"
Got to "Updates" -> "update settings" select "stable"

on 15.04 Devel (285) and that did not fix the issue.

**Comment by necoli1822** [comment]
Thank you for adding more cases.
I’ve also tried to change channels, searching the error-some codes, but didn’t work.
To specify my symptom, it lists the actual carriers available in Korea, but somewhat outdated after minutes. (up to 5 minutes; but when I select, it doesn’t show any changes.)

**Comment by LightofAeons** [comment]
Hi @necoli1822 

That is the exact same issue that I'm having at the moment.

Out of interest, did this recently occur, or have you always been unable to register your SIM card's carrier with your phone?

**Comment by necoli1822** [comment]
Domo @LightofAeons

I’m doubt whether it is a problem of outdated carrier list.

I’ve recently changed my phone to Nexus 5, due to the old nexus 4 excluded from orthodox support. It happened since I first installed touch from Ubport-installer a month ago.

**Comment by LightofAeons** [comment]
It could very well be an outdated carrier list :D !

I will also try installing 16.04 Devel when I get home and report on that (I had no probem at all using my SIM with stock Android).

If that doesn't work, I will reinstall stock Android and then switch again to 15.04 Stable and try again.

**Comment by LightofAeons** [comment]
Okay, updated to 16.04/devel and it didn't even recognise my SIM card.

Switched back to Stock Android (Hammerhead-m4b30z) and SIM connected successfully to the network.

Reinstalled 15.04/stable and the phone can't even find a carrier this time round.

**Comment by LightofAeons** [comment]
I have now tried switching from Stock Android (Hammerhead-m4b30z) to 15.04/stable, no carrier recognised. I repeated the same process for 15.04/devel (again no carrier recognised) and 16.04/devel (no mobile sim recognise ).

I'm not too sure what else to try in that case.

**Comment by necoli1822** [comment]
It might be same error with #376 rather than #377 .
Maybe, we need to wait for the next update.

**Comment by LightofAeons** [comment]
Yep, I saw that issue as well! I have posted this issue in the Telegram supergroup and Welcome groups and am currently waiting for a reply.

**Comment by LightofAeons** [comment]
Here is the reply from Lionelb on the welcome group:

> 
I had something similar with Canonical years ago. The fix was non-trivial and involved adaptations to strings, using the terminal. This sounds even more complicated, owing to the additional security layers. Did you experiment with SIMs from other providers?

I replied that I had not tried other SIMs.

**Comment by necoli1822** [comment]
In my case, I have tried 2 sims, most famous and majour carriers in Korea(~65% of all sim users would be either one of them) and failed.
Thus, it might be affect both of Korea and Japan, or other non-UT-friendly countries in my opinion.

**Comment by LightofAeons** [comment]
Yep, you've hit the nail on the head there man! NTT Docomo is also one of the main providers here. Not to worry, hopefully there will be an update in the near future that fixes the issue.

**Comment by necoli1822** [comment]
@NeoTheThird 
I`ve tried to install and update from stable, rc, and dev, but totally failed to make my Nexus5 work with the sim.
Here are the UTlogcat and UTdmesg to ask for further help.

[UTdmesg.txt](https://github.com/ubports/ubuntu-touch/files/1696781/UTdmesg.txt)
[UTlogcat.txt](https://github.com/ubports/ubuntu-touch/files/1696782/UTlogcat.txt)

**Comment by necoli1822** [comment]
What I have done to make those file is
1. Reboot the phone.
2. Go into Setting
3. Go into Change Carrier.
4. Wait until the list comes up.
5. Choose my carrier.
6. Wait until it fails.
7. Nothing happened and turn the screen off.

**Comment by SaltyCybernaut** [comment]
This looks to be related to issue [#376](https://github.com/ubports/ubuntu-touch/issues/376) and [this](https://github.com/ubports/ubuntu-touch/issues/376#issuecomment-380312902) is how I was able to fix my issue and it might resolve this issue as well.

**Comment by fw190d13** [comment]
I can confirm this. I can't change from 3G to 2G. After tapping the option only a circle is shown and it will just spin without showing the menu to change the network type.

**Comment by schnuckelchen** [comment]
Similar issue: Nexus 5, rc-branch, provider simplytel (e-net, germany).
If the signal is weaker, the phone will not connect to the provider. The provider is displayed (On Settings > Cellular Network > Carrier and APN > Carrier > ...) but not selected.

**Comment by ITwrx** [comment]
@schnuckelchen same with me. nexus 5, 16.04 stable  and Ting carrier. When first setting up phone with weak signal "automatically" is the default option for carrier setting. i went outside to get a better signal and it still wouldn't connect automatically. 4 or 5 Ting carriers were listed in carrier settings area (for some reason) and "automatically" was still selected. no data. manually selected the first "Ting" and it connected right up.

**Comment by schnuckelchen** [comment]
The problem is solved for my device.
I opened it and found that one of the two antennas was broken. In addition, the motherboard was broken.
I repaired both and have no problem with the signal.

**Comment by ITwrx** [comment]
In my case, my weak signal had nothing to do with my phone, nor UBports. i was just chiming in b/c the way carrier selection is handled seems to be a little buggy, for me at least, especially in concert with a weak signal.
